Data/Vector/Generic.hs

```diff
@@ -432,6 +434,14 @@ splitAt n v = ( unsafeSlice 0 m v
       n'  = max n 0
       len = length v
 
+-- | /O(1)/ Yield the 'head' and 'tail' of the vector, or 'Nothing' if empty.
```

Comment:
why aren't we thinking about how to adapt the fusion codes for head/tail as found in Stream.Monadic and/or bundle?

https://github.com/haskell/vector/blob/master/Data/Vector/Fusion/Stream/Monadic.hs#L341
for one of them